joomla/joomla-cms
# [4.0] Automate the calendar locales generation

Pull Request for Issue # .

### Summary of Changes
When the new calendar was introduced all the translatable strings were put in a js file. This was required to assure that the calendar would work correctly form the first release. Also this caused some problems to the translation teams as the had to deal with js files apart from the known ini.

So this PR automates the creation of these js files by:
- placing the relative strings in the installation folder (where all known languages exist)
- update the languages xmls with all the needed values (also some comments so ppl can understand what these values stand for)
- ATM the old js files are still in place and the generated ones are placed in a tmp folder in `media/system/js/fields` (for validating the procedure)

### Testing Instructions
Fork my brach,
run `npm i`
run `node build.js --installer`
Check that the generated files in `media/system/js/fields/tmp` are the same as the ones in `media/system/js/fields/calendar-locales`

PS only two files are generated right now as the installation contains only en-GB and en-US

### Expected result

![screen shot 2018-02-24 at 15 12 53](https://user-images.githubusercontent.com/3889375/36631272-3e8ee052-1975-11e8-8e42-0559584ed2ed.png)

### Documentation Changes Required
Yes

@Bakual @infograf768 @dneukirchen @wilsonge @mbabker
